File: dynamic_listings/python/default/collisiondetection/AABB.py
class Point:

    """A simple Point"""

    def __init__(self, x: int, y: int) -> None:
        """
        Creates the point
        :x: The X Coordinate of the point
        :y: The Y Coordinate of the point
        """
        self.x: int = x
        self.y: int = y


class Rectangle:

    """A rectangle, made with an upper-left corner, width and height"""

    def __init__(self, point: Point, width: int, height: int) -> None:
        """Creates the rectange

        :point: The upper left corner point of the rectangle
        :width: The rectangle Width
        :height: The rectangle Height

        """
        self.corner: Point = point
        self.width: int = width
        self.height: int = height


def rect_rect_collision(A: Rectangle, B: Rectangle) -> bool:
    """
    Checks for an axis-aligned bounding rectangle collision

    :A: The First Rectangle
    :B: The Second Rectangle
    :returns: A boolean that tells if the rectangles collided
    """
    
    is_colliding = (A.corner.x < B.corner.x + B.width) and\
       (A.corner.x + A.width > B.corner.x) and\
       (A.corner.y < B.corner.y + B.height) and\
       (A.corner.y + A.height > A.corner.y)
    
    return is_colliding

File: dynamic_listings/rust/default/collisiondetection/AABB.rs
# #![allow(dead_code)]
struct Point {
    // Rewritten as a memo
    x: u32,
    y: u32,
}

struct Rectangle {
    corner: Point,
    width: u32,
    height: u32,
}

fn rect_rect_collision(a: &Rectangle, b: &Rectangle) -> bool {
    a.corner.x < b.corner.x + b.width
        && a.corner.x + a.width > b.corner.x
        && a.corner.y < b.corner.y + b.height
        && a.corner.y + a.height > a.corner.y
}
